### PR TITLE
fix(#2089): visual fix for interactive states on input

### PR DESCRIPTION
**Pull request by twjeffery**
This is to fix this bug: https://github.com/GovAlta/ui-components/issues/2089

# Before (the change)
image

The visual appearance of the error state on form fields is not consistent between Figma and Code components:
- Red border is 2px vs. 3px.
- Red border is set to Inside vs. Outside.
- Fields with leading /trailing content have no red border around prefix/suffix vs. border around entire field plus thicker border around input field.
- hover on default inputs (without trailing/leading content) hover border is outside - should be inside
- should have no hover state on focus 
- should have no hover state on error
- should not show borders around leading and trailing content

# After (the change)
image
image
image
image
image

- Hover border is interior `--goa-border-width-m`
- Error border is interior `--goa-border-width-m`
- Interactive border is exterior `--goa-border-width-l`
- Error state has no hover state
- Focus state has no hover state
- goa variables used everywhere applicable
- leading and trailing content has no border, only input

### CODE REVIEW

**Comment by lizhuomeng71** [comment]
There are few thing we have noticed with this change

1) Error State with Leading and Trailing Content missing border left and border right

image

2) Error state with Leading and Trailing content hover state will change border color, while it stated on the ticket that Error state has no hover state

**Comment by twjeffery** [comment]
> There are few thing we have noticed with this change
> 
> 1. Error State with Leading and Trailing Content missing border left and border right
> 
> image
> 
> 2. Error state with Leading and Trailing content hover state will change border color, while it stated on the ticket that Error state has no hover state
> 
> This happen for both angular and react

@lizhuomeng71 I couldn't figure out how to achieve what you noted above. The border style wraps around the leading and trailing icon, which makes it hard to easily change. The major issues are fixed, these are minor issues on edge cases.

**Comment by lizhuomeng71** [comment]
This Main Ticket is fixed, the two edge case will be created new ticket separately

File: libs/web-components/src/components/input/Input.svelte [hunk 1]
```diff
@@ -387,28 +412,32 @@
...
+    margin-right: var(--goa-space-s);
   }
 
   input {
     display: inline-block;
     color: var(--goa-color-text-default);
     font-size: var(--goa-font-size-4);
-    padding: var(--goa-space-xs) var(--goa-space-s);
+
+    /* why are their calcs? what are they trying to fix? */
```
[the remark this answers is not in the document]
Comment (twjeffery) [reply libs/web-components/src/components/input/Input.svelte]: removed

**Comment by twjeffery** [comment]
@dustin-nielsen-goa error border not showing up on leading and trailing content is now fixed as well (https://github.com/GovAlta/ui-components/issues/2235)

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**